src/components/Accordion/Accordion.js

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import AccordionItem from './AccordionItem';
+
+const CN = 'accordion';
+
+class Accordion extends Component {
+  constructor(props) {
+    super(props);
+
+    this.clickHandler = this.clickHandler.bind(this);
+    this.state = {
+      hide: [],
+      currentIndex: null,
+    };
+  }
+
+  componentDidMount() {
+    const { data, open } = this.props;
+    const hide = data.map((el) => !open);
+
+    this.setState({ hide: hide });
+  }
+
+  clickHandler(index) {
+    const { hide } = this.state;
+    const { showAll } = this.props;
+    let hideCopy = [...hide];
+
+    hideCopy[index] = !hideCopy[index];
+
+    if (!showAll) {
+      hideCopy = hideCopy.map((el, currentIndex) =>
+        index === currentIndex ? el : true,
```

Comment:
Could you please describe the logic here? I can't get it from the code

Reply:
Logic of all this function's work

Reply:
I take an array of 'hide' values for all accordion items and change the 'hide' value of  the clicked item to the opposite and  
1)If showAll prop is true, then 'hide' value of the rest items is not changed
2)If showAll prop is false, then 'hide' value of the rest items is set to true